distributed/client.py

```diff
@@ -3481,6 +3482,101 @@ def scheduler_info(self, **kwargs):
             self.sync(self._update_scheduler_info)
         return self._scheduler_identity
 
+    async def _dump_cluster_state(
+        self,
+        filename: str,
+        exclude: Sequence[str] = None,
+        format: Literal["msgpack"] | Literal["yaml"] = "msgpack",
+    ) -> None:
+
+        scheduler_info = self.scheduler.dump_state()
+
+        worker_info = self.scheduler.broadcast(
+            msg=dict(
+                op="dump_state",
+                exclude=exclude,
+            ),
+        )
+
+        scheduler_info, worker_info = await asyncio.gather(scheduler_info, worker_info)
+        state = {
+            "scheduler_info": scheduler_info,
+            "worker_info": worker_info,
+        }
+        if format == "msgpack":
+            import gzip
+
+            import msgpack
+            import yaml
+
+            with gzip.open(filename, "wb") as fdg:
+                msgpack.pack(state, fdg)
+        elif format == "yaml":
+            import yaml
+
+            with open(filename, "w") as fd:
+                yaml.dump(state, fd)
+        else:
+            raise ValueError(
+                f"Unsupported format {format}. Possible values are `msgpack` or `yaml`"
+            )
+
+    def dump_cluster_state(
+        self,
+        filename: str,
```

Comment:
With other methods that generate local files, e.g. `performance_report`, `get_task_stream`, I've found providing a default filename to be convenient. In this case, it would also let us encode the default `format` in the filename (when playing around with this PR locally I was confused about this and tried to open a file with `yaml`) 

```suggestion
        filename: str = "cluster-dump.msgpack",
```